File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer, honors;
    
    for(int i = 0; i < score.size(); i++) {
        if(i < k) {
            honors.push_back(score[i]);
        } else {
            honors[k - 1] = honors[k - 1] < score[i] ? score[i] : honors[k - 1];
        }
        
        sort(honors.rbegin(), honors.rend());
        int temp = i < k ? honors[i] : honors[k-1];
        answer.push_back(temp);
    }
    
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cs
using System;
using System.Linq;

public class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.Length];
        int[] honors = new int[k];
        
        for(int i = 0; i < score.Length; i++) {
            honors[k - 1] = honors[k - 1] < score[i] ? score[i] : honors[k - 1];
            honors = honors.OrderByDescending(n => n).ToArray();
            answer[i] = i < k ? honors[i] : honors[k-1];
        }
        
        return answer;
    }
}